File: app/src/main/java/za/co/jpsoft/winkerkreader/NavigationHandler.java
package za.co.jpsoft.winkerkreader;

import android.widget.TextView;

import static za.co.jpsoft.winkerkreader.data.WinkerkContract.winkerkEntry;

/**
 * Handles navigation between different data views via swipe gestures
 */
public class NavigationHandler {

    public static void handleLeftSwipe(MainActivity2 activity, TextView sortOrderView) {
        switch (winkerkEntry.SORTORDER) {
            case "HUWELIK":
                switchTo(activity, sortOrderView, "VAN", "VAN");
                break;
            case "VAN":
                switchTo(activity, sortOrderView, "GESINNE", "GESINNE");
                break;
            case "GESINNE":
                switchTo(activity, sortOrderView, "WYK", "WYK");
                break;
            case "WYK":
                switchTo(activity, sortOrderView, "OUDERDOM", "OUDERDOM");
                break;
            case "OUDERDOM":
                switchTo(activity, sortOrderView, "ADRES", "ADRES");
                break;
            case "ADRES":
                switchTo(activity, sortOrderView, "VERJAAR", "VERJAAR");
                break;
            case "VERJAAR":
                switchTo(activity, sortOrderView, "HUWELIK", "HUWELIK");
                break;
        }
    }

    public static void handleRightSwipe(MainActivity2 activity, TextView sortOrderView) {
        switch (winkerkEntry.SORTORDER) {
            case "HUWELIK":
                switchTo(activity, sortOrderView, "VERJAAR", "VERJAAR");
                break;
            case "VERJAAR":
                switchTo(activity, sortOrderView, "ADRES", "ADRES");
                break;
            case "ADRES":
                switchTo(activity, sortOrderView, "OUDERDOM", "OUDERDOM");
                break;
            case "OUDERDOM":
                switchTo(activity, sortOrderView, "WYK", "WYK");
                break;
            case "WYK":
                switchTo(activity, sortOrderView, "GESINNE", "GESINNE");
                break;
            case "GESINNE":
                switchTo(activity, sortOrderView, "VAN", "VAN");
                break;
            case "VAN":
                switchTo(activity, sortOrderView, "HUWELIK", "HUWELIK");
                break;
        }
    }

    private static void switchTo(MainActivity2 activity, TextView sortOrderView, String sortOrder, String layout) {
        sortOrderView.setBackground(null);
        winkerkEntry.SORTORDER = sortOrder;
        winkerkEntry.DEFLAYOUT = layout;
        winkerkEntry.SOEKLIST = false;
        activity.observeDataset();
    }
}

File: app/src/main/kotlin/NavigationHandler.kt
// NavigationHandler.kt
package za.co.jpsoft.winkerkreader

import android.widget.TextView
import za.co.jpsoft.winkerkreader.data.WinkerkContract.winkerkEntry

/**
 * Handles navigation between different data views via swipe gestures
 */
object NavigationHandler {

    @JvmStatic
    fun handleLeftSwipe(activity: MainActivity2, sortOrderView: TextView) {
        when (winkerkEntry.SORTORDER) {
            "HUWELIK" -> switchTo(activity, sortOrderView, "VAN", "VAN")
            "VAN" -> switchTo(activity, sortOrderView, "GESINNE", "GESINNE")
            "GESINNE" -> switchTo(activity, sortOrderView, "WYK", "WYK")
            "WYK" -> switchTo(activity, sortOrderView, "OUDERDOM", "OUDERDOM")
            "OUDERDOM" -> switchTo(activity, sortOrderView, "ADRES", "ADRES")
            "ADRES" -> switchTo(activity, sortOrderView, "VERJAAR", "VERJAAR")
            "VERJAAR" -> switchTo(activity, sortOrderView, "HUWELIK", "HUWELIK")
        }
    }

    @JvmStatic
    fun handleRightSwipe(activity: MainActivity2, sortOrderView: TextView) {
        when (winkerkEntry.SORTORDER) {
            "HUWELIK" -> switchTo(activity, sortOrderView, "VERJAAR", "VERJAAR")
            "VERJAAR" -> switchTo(activity, sortOrderView, "ADRES", "ADRES")
            "ADRES" -> switchTo(activity, sortOrderView, "OUDERDOM", "OUDERDOM")
            "OUDERDOM" -> switchTo(activity, sortOrderView, "WYK", "WYK")
            "WYK" -> switchTo(activity, sortOrderView, "GESINNE", "GESINNE")
            "GESINNE" -> switchTo(activity, sortOrderView, "VAN", "VAN")
            "VAN" -> switchTo(activity, sortOrderView, "HUWELIK", "HUWELIK")
        }
    }

    private fun switchTo(activity: MainActivity2, sortOrderView: TextView, sortOrder: String, layout: String) {
        sortOrderView.background = null
        winkerkEntry.SORTORDER = sortOrder
        winkerkEntry.DEFLAYOUT = layout
        winkerkEntry.SOEKLIST = false
        activity.observeDataset()
    }
}